vm-setup/roles/v1aX_integration_test/tasks/inspection.yml

```diff
@@ -0,0 +1,58 @@
+---
+  - name: Install jmespath
+    pip:
+      executable: "pip3"
+      name: jmespath
+      state: present
+    become: yes
+    become_user: root
+
+  - name: Wait until "{{ NUM_NODES }}" bmhs become ready.
+    shell: |
+        kubectl get bmh -n {{ NAMESPACE }} -o json | jq -r '[ .items[]
+        | select (.status.provisioning.state == "ready")
+        | .metadata.name ] | length'
+    register: ready_bmhs
+    retries: 150
+    delay: 10
+    until: ready_bmhs.stdout == NUM_NODES
+
+  - name: Get a list of all BareMetalHosts
+    community.kubernetes.k8s_info:
+      api_version: metal3.io/v1alpha1
+      kind: BareMetalHost
+      namespace: metal3
+    register: bmh_list
+
+  - name: Request inspection for hosts via API
+    community.kubernetes.k8s:
+      state: present
+      definition:
+        apiVersion: metal3.io/v1alpha1
+        kind: BareMetalHost
+        metadata:
+          name: "{{  item  }}"
+          namespace: "{{ NAMESPACE }}"
+          annotations:
+            inspect.metal3.io: ""
+    loop: "{{ bmh_list | json_query('resources[*].metadata.name') }}"
+
+  - name: Wait until "{{ NUM_NODES }}" bmhs are in inspecting or inspect wait state.
+    shell: |
+        kubectl get bmh -n {{ NAMESPACE }} -o json | jq -r '[ .items[]
+        | select (.status.provisioning.state == "inspecting" or .status.provisioning.state == "inspect wait")
```

Comment:
Some related questions, after reading ref. proposal: 

- I was not able to find the mention in the proposal stating BMH might be in  `inspect wait` state while it is inspected, so why we should consider that state? 
- Is it possible to check once the inspection is finished, whether the previous inspection data is updated on the spec of the BareMetalHost object or that is out of scope of this test case?

Reply:
> Some related questions, after reading ref. proposal:
> 
> * I was not able to find the mention in the proposal stating BMH might be in  `inspect wait` state while it is inspected, so why we should consider that state?

When node is being inspected, it will have either inspecting or inspect wait. This is to ensure that inspection was really triggered. 

> * Is it possible to check once the inspection is finished, whether the previous inspection data is updated on the spec of the BareMetalHost object or that is out of scope of this test case?

It would be nice to have that sort of check, but I'm not sure how to do it. Any suggestions?

Reply:
Not sure either if that is possible to check, so that's why was asking if current logic is already enough to say that BMH spec is already updated with new HW details.